src/Domain/FeedbackPostDomain.php

```diff
@@ -0,0 +1,99 @@
+<?php
+namespace Wheniwork\Feedback\Domain;
+
+use Wheniwork\Feedback\Service\Authorizer;
+use Wheniwork\Feedback\Service\GithubService;
+use Wheniwork\Feedback\Service\HipChatService;
+
+abstract class FeedbackPostDomain extends FeedbackDomain
+{
+    private $auth;
+
+    public function __construct(
+        HipChatService $hipchat,
+        GithubService $github,
+        Authorizer $auth
+    ) {
+        parent::__construct($hipchat, $github);
+        $this->auth = $auth;
+    }
+
+    public function __invoke(array $input)
+    {
+        $payload = $this->getPayload();
+
+        try {
+            $this->auth->ensure($input);
+
+            foreach ($this->getRequiredFields() as $field) {
+                if (empty($input[$field])) {
+                    throw new \RuntimeException("Missing required field '$field'");
```

Comment:
This is okay but does have a downside in only showing a single error at a time. Often I do something like this:

```php
$missing = array_diff($this->getRequiredFields(), array_keys($input));
if ($missing) {
    throw new RuntimeException(
        'Missing some required values: ' .
        implode(', ', $missing)
    );
}
```

This reduces the number of times a person has to call the code to see multiple missing values.